packages/core/admin/admin/src/components/LeftMenu.tsx

```diff
@@ -1,27 +1,47 @@
 import * as React from 'react';
 
 import { Divider, Flex, FlexComponent } from '@strapi/design-system';
-import { Feather, Lock, House } from '@strapi/icons';
+import { Lock } from '@strapi/icons';
 import { useIntl } from 'react-intl';
 import { useLocation } from 'react-router-dom';
 import { styled } from 'styled-components';
 
 import { useAuth } from '../features/Auth';
 import { useTracking } from '../features/Tracking';
-import { Menu } from '../hooks/useMenu';
+import { Menu, MenuItem } from '../hooks/useMenu';
 import { getDisplayName } from '../utils/users';
 
 import { MainNav } from './MainNav/MainNav';
 import { NavBrand } from './MainNav/NavBrand';
 import { NavLink } from './MainNav/NavLink';
 import { NavUser } from './MainNav/NavUser';
 
-const NewNavLinkBadge = styled(NavLink.Badge)`
+const sortLinks = (links: MenuItem[]) => {
+  return links.sort((a, b) => {
+    // if no position is defined, we put the link in the position of the external plugins, before the plugins list
+    const positionA = a.position ?? 6;
+    const positionB = b.position ?? 6;
+
+    if (positionA === positionB) {
+      return (a.intlLabel.defaultMessage ?? '') < (b.intlLabel.defaultMessage ?? '') ? -1 : 1;
```

Comment:
theres a couple of issues with this, I would firstly run `formatMessage` on all these links, you could do a map before hand, secondly you want to use `useCollator` because this won't account for certain characters in certain languages 👍🏼